### PR TITLE
MM-51585 : Fix duplicated date separator in center channel when Pinned posts RHS is open

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
This PR fixes an issue where date separator is shown in the center channel for every post when Pinned Posts RHS is open

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/22950
https://mattermost.atlassian.net/browse/MM-51585

#### Screenshots
|  before  |  after  |
|----|----|
| Screen Shot 2023-04-20 at 22 48 12 |Screen Shot 2023-04-20 at 22 47 04|

#### Release Note
```release-note
Resolved date separator display in center channel posts when Pinned Posts RHS is open
```

### CODE REVIEW

File: webapp/channels/src/components/post/post_component.tsx [hunk 1]
```diff
@@ -473,7 +473,7 @@ const PostComponent = (props: Props): JSX.Element => {
             className={props.location === 'SEARCH' ? 'search-item__container' : undefined}
             data-testid={props.location === 'SEARCH' ? 'search-item-container' : undefined}
         >
-            {(isSearchResultItem || props.isPinnedPosts || props.isFlagged) && <DateSeparator date={currentPostDay}/>}
+            {(isSearchResultItem || (props.location !== Locations.CENTER && props.isPinnedPosts) || props.isFlagged) && <DateSeparator date={currentPostDay}/>}
```
Comment (Apetree100122) [inline webapp/channels/src/components/post/post_component.tsx:476]: #23073

**Review by nevyangelova — APPROVED** [review]
LGTM!

**Comment by M-ZubairAhmed** [comment]
@sinansonmez is this issue not happening in RHS?

**Comment by sinansonmez** [comment]
@M-ZubairAhmed  As you can see in the screenshot, in RHS every post has a seperator. I thought this is the expected behavior for displaying pinned posts in RHS

**Comment by M-ZubairAhmed** [comment]
i am going to quickly rope in @itao for confirm the RHS side of behaviour

**Review by johndavidlugtu — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by yasserfaraazkhan — APPROVED** [review]
@M-ZubairAhmed  have run few tests locally since e2e was timing out
Screenshot 2023-06-01 at 10 08 13 PM

**Comment by M-ZubairAhmed** [comment]
@yasserfaraazkhan the CI for E2E seems to be fixed now, lets try running these again :)

**Comment by yasserfaraazkhan** [comment]
@M-ZubairAhmed this should be good to merge :D

**Pull request merged**